### DISCUSSION TITLE
How best to manage binary python package dependencies

### DISCUSSION THREAD

**Discussion by masaccio** [Homebrew/discussions#5670] [Writing Formulae/Casks]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

I'm looking to add a python package of mine to Homebrew and I am tumbling for a rabbit hole regarding dependencies.

My current Formula [lives here](https://github.com/masaccio/homebrew-numbers-parser/blob/main/Formula/numbers-parser.rb) and this started using a bunch or resources that are source distributions.

The problem with this is that I ultimately need to pull in some large tooling dependencies to build all these packages from source (such as Rust). This feels like overkill to me and the dependency list would be quite large requiring users to install lots of formulas just for a python package.

My next idea was to explicitly list the packages from PyPI that are the dependencies, but some of these are binary packages so now I am looking at the complexity of architecture detection.

Pip figures this all out of the user and downloads the appropriate wheels. How is this achieved in Homebrew formulas?

**Comment by SMillerDev** [comment]
> The problem with this is that I ultimately need to pull in some large tooling dependencies to build all these packages from source (such as Rust). This feels like overkill to me and the dependency list would be quite large requiring users to install lots of formulas just for a python package.

This is intentional, since we want Homebrew CI to build an optimised binary. If you have a resource for something like `cryptography`, you can use a `depends_on` with https://formulae.brew.sh/formula/cryptography so you don't need to depend on rust for that.

**Comment by masaccio** [comment]
`cramjam` is the main culprit that needs Rust, Cmake and autotools at least. All things a native developer might want from Homebrew but not many a Python dev or consumer of command line tools. I’ll add contributing a cramjam formula to my todo list. Thanks for clarifying the project’s intent.
